moveos/moveos-stdlib/moveos-stdlib/sources/events.move

```diff
@@ -10,14 +146,58 @@ module moveos_std::events {
     ///
     /// The type T is the main way to index the event, and can contain
     /// phantom parameters, eg emit(MyEvent<phantom T>).
-    public fun emit_event<T: drop + store>(ctx: &mut TxContext, event: T) {
-        let guid = tx_context::fresh_object_id(ctx);
-
-        //!Notice that: count now just for For placeholder, and value is always 0
-        write_to_event_store<T>(&guid, 0,  event);
+    public fun emit_event<T: key>(ctx: &mut StorageContext, event: T) {
```

Comment:
Should we require the `T` to be `private_generics`? Help record a security issue for this, and we can figure it out in the future.